modules/core/src/copy.cpp

```diff
@@ -411,6 +411,8 @@ Mat& Mat::operator = (const Scalar& s)
 {
     CV_INSTRUMENT_REGION()
 
+    if (empty()) return *this;
```

Comment:
IMHO, there should be an exception. See [this code](https://github.com/opencv/opencv/pull/11984/commits/c473718bc24e1b6197696cb22a2ca46f2c6a1c3b#diff-2271efc7b6d06fd8b0b6d04a723898fcL92).